### PR TITLE
Elasticsearch: Fix handling multiple datalinks for a single field

### LINKED ISSUE #43521: [ES] [data-links in ConfigEditor] adding multiple data links on same field in datasource settings doesn't work
**What happened**:
image
image
adding multiple data links on same field in datasource settings don't work
**What you expected to happen**:
there are multiple button after field which bind multple data links
**How to reproduce it (as minimally and precisely as possible)**:
add multiple data links on same field
**Anything else we need to know?**:

**Environment**:
- Grafana version: v8.3.0
- Data source type & version: elasticsearch
- OS Grafana is installed on: linux
- User OS & Browser: windows chrome
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by Elfo404**
**What this PR does / why we need it**:
As described in the linked issue, when adding multiple datalinks for a single field in Elasticsearch, only the first one was being shown. This PR changes this behavior by ensuring all configured links are added for each field.

**Which issue(s) this PR fixes**:

Fixes #43521

**How to test this:**

Using elastic77 devenv (or any other ES version) create a new datasource and configure as this:
Screenshot 2022-01-14 at 10 08 11

then create multiple datalinks for the same field, they can be both external, internal or a mix of both:
Screenshot 2022-01-14 at 10 10 14

when then visualizing logs in explore, both links should show up next to the matching field:
Screenshot 2022-01-14 at 10 10 51

### CODE REVIEW

File: public/app/plugins/datasource/elasticsearch/datasource.ts [hunk 1]
```diff
@@ -936,49 +936,49 @@ export class ElasticDatasource
...
-
   if (limit) {
     dataFrame.meta = {
       ...dataFrame.meta,
       limit,
     };
   }
 
-  if (!dataLinks.length) {
+  if (!dataLinks.length || !dataLinks) {
```
Comment (ifrost) [inline public/app/plugins/datasource/elasticsearch/datasource.ts:946]: Is it needed? I think `dataLinks` are always defined (initialized in the [constructor](https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/elasticsearch/datasource.ts#L113))

File: public/app/plugins/datasource/elasticsearch/datasource.ts [hunk 2]
```diff
@@ -936,49 +936,49 @@ export class ElasticDatasource
...
     return;
   }
 
   for (const field of dataFrame.fields) {
-    const dataLinkConfig = dataLinks.find((dataLink) => field.name && field.name.match(dataLink.field));
+    const linksToApply = dataLinks.filter((dataLink) => new RegExp(dataLink.field).test(field.name));
```
Comment (ifrost) [inline public/app/plugins/datasource/elasticsearch/datasource.ts:951]: 👍 🚀

**Review by ifrost — APPROVED** [review]
Nice one!

File: public/app/plugins/datasource/elasticsearch/datasource.ts (hunk 1 above)
Comment (Elfo404) [reply public/app/plugins/datasource/elasticsearch/datasource.ts:946]: i have no idea why i added it 🤔

**Pull request merged**